backend/api/view/StreamPostsView.py

```diff
@@ -23,68 +32,104 @@ def get(self, request):
+        #         friend_list_data = get_local_friends_list(user_id)
+        #
+        #     authors_followed = Follow.objects.filter(authorA=user_id)
+        #
+        #     foreign_hosts = []
+        #     for author in authors_followed:
+        #         author_uuid = get_author_profile_uuid(author.authorB)
+        #         try:
+        #             author_profile = AuthorProfile.objects.get(id=author_uuid)
+        #             query_set = query_set | Post.objects.filter(author=author_profile)
+        #         except:
+        #             # add foreign host to list to get it latter
+        #             parsed_url = urlparse(author.authorB)
+        #             author_host = '{}://{}/'.format(parsed_url.scheme, parsed_url.netloc)
+        #             if author_host not in foreign_hosts:
+        #                 foreign_hosts.append(author_host)
+        #
+        #     query_set = query_set.order_by("-published")
+        #     stream_posts = PostSerializer(query_set, many=True).data
+        #     stream = []
+        #     for post in stream_posts:
+        #         if (can_read(user_id, post, friend_list_data)):
+        #             stream.append(post)
+        # except:
+        #     return Response("Author does not exist", status.HTTP_400_BAD_REQUEST)
+        #
+        # # if request is not forwarded
+        # if not ServerUser.objects.filter(user=request.user).exists():
+        #     for foreign_host in foreign_hosts:
+        #         try:
+        #             server_user = ServerUser.objects.get(host=foreign_host)
+        #             headers = {'Content-type': 'application/json',
+        #                        "X-Request-User-ID": AuthorProfileSerializer(user_profile).data["id"]}
+        #             url = "{}{}author/posts".format(server_user.host, server_user.prefix)
+        #             response = requests.get(url,
+        #                                     auth=(server_user.send_username, server_user.send_password),
+        #                                     headers=headers)
+        #
+        #             if response.status_code == 200:
+        #                 response_json = json.loads(response.content)
+        #                 stream += response_json["posts"]
+        #         except ServerUser.DoesNotExist:
+        #             return Response("Get request fail, bad foreign host",
+        #                             status.HTTP_400_BAD_REQUEST)
+        #         except Exception as e:
+        #             pass
+        #             # return Response("Error: Get foreign author post failed", status.HTTP_400_BAD_REQUEST)
+        #
+        # sorted_stream = sorted(stream, key=lambda k: k['published'], reverse=True)
+        #
+        # response_data = {
+        #     "query": "posts",
+        #     "count": len(sorted_stream),
+        #     "posts": sorted_stream
+        # }
+        #
+        # return Response(response_data, status.HTTP_200_OK)
```

Comment:
Why is there commented out code here?

Reply:
just in case we want to go back to the version that only return posts that we are following